src/newTest/java/org/opensearch/test/framework/cluster/OpenSearchClientProvider.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2020 floragunn GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.test.framework.cluster;
+
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyStore;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.apache.http.Header;
+import org.apache.http.message.BasicHeader;
+import org.opensearch.security.support.PemKeyReader;
+import org.opensearch.test.framework.certificate.TestCertificates;
+
+
+public interface OpenSearchClientProvider {
```

Comment:
Is this interface needed? It doesn't have any documentation

Reply:
Usually, the tests will retrieve the client directly from the cluster. Thus, the cluster implements this interface. However, in some special tests, we may need to consider specific nodes and retrieve clients from these. Thus, also the nodes would implement that interface.

Reply:
I'd rather remove the interface and we can add it when it's generating value for the codebase.

Reply:
Right now, the interface also has some value by providing separation of concerns.  It separates the internal management of the cluster (located in `LocalCluster` and `LocalOpenSearchCluster.Node`) from the logic necessary to create clients for the cluster.

Integrating the code would increase the complexity of  `LocalCluster`.